Add setting that allows users to hide/show dock icon

timlenardo opened the pull request:
Requested in #316  - this PR allows users to hide/show the dock icon: 

https://github.com/user-attachments/assets/893f0c9d-d130-43fc-b0a4-7701b3e7058d

Under the hood, this toggles whether the app's 'ActivationPolicy' is in .accessory or .regular mode. 

I am concerned that switching it to .accessory will have some unforeseen consequences. @Niduank - you handled switching us from .accessory to .regular before. Did you have to change anything to make it work? 

 We should all use the app in accessory mode for a while to ensure everything is working correctly before releasing it.

Niduank commented:
@timlenardo 
Setting the app to .accessory only affects its appearance in:
- The Dock,
- CMD+Tab,
- And the top-left menu bar.

Back then, I had to add the "LSUIElement" key in the Info.plist to force the app into accessory mode but `NSApp.setActivationPolicy` is actually enough!

The PR looks good, we can avoid code duplication and make a `static func toggleDockIconVisibility` in `App` or `AppDelegate`

Niduank commented on macos/Onit/AppDelegate.swift:
```diff
@@ -37,4 +42,18 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         return true
     }
+    
+    private func configureDockIconVisibility() {
```
In order to avoid code duplication:
```swift
static func toggleDockIconVisibility()
``` 
and call in GeneralTab : `AppDelegate.toggleDockIconVisibility()`

timlenardo replied:
Okay, I'll try this. I started with something similar but was having casting issues. The 'AppDelegate' available in views like GeneralTab is a generic instance, rather than our specific override. I think using a static function, as you suggest, should solve it though.

timlenardo replied:
Making the function static worked well, good suggestion @Niduank 👍

timlenardo commented:
In addition to addresses the comments, I also decided to move the settings toggle from the "Auto-start" section (where it didn't make much sense), into the Display section. Ready to merge now!
